hazelcast/src/main/java/com/hazelcast/internal/bootstrap/TpcServerBootstrap.java

```diff
@@ -152,22 +172,44 @@ private void openServerSockets() {
 
     // public for testing
     public AltoSocketConfig getClientSocketConfig() {
-        if (nodeEngine.getConfig().getAdvancedNetworkConfig().isEnabled()) {
-            ServerSocketEndpointConfig endpointConfig = (ServerSocketEndpointConfig) nodeEngine
-                    .getConfig()
+        validateSocketConfig();
+
+        if (config.getAdvancedNetworkConfig().isEnabled()) {
+            ServerSocketEndpointConfig endpointConfig = (ServerSocketEndpointConfig) config
                     .getAdvancedNetworkConfig()
                     .getEndpointConfigs()
                     .get(EndpointQualifier.CLIENT);
 
-            if (endpointConfig == null) {
-                return null;
-            }
-
             return endpointConfig.getAltoSocketConfig();
         }
 
         // unified socket
-        return nodeEngine.getConfig().getNetworkConfig().getAltoSocketConfig();
+        return config.getNetworkConfig().getAltoSocketConfig();
+    }
+
+    private void validateSocketConfig() {
+        AdvancedNetworkConfig advancedNetworkConfig = config.getAdvancedNetworkConfig();
```

Comment:
I don't believe it should be the responsibility of this class to verify the configuration. This should be done as part of the configuration loading process.

Reply:
Not possible, socket config isn't aware where it belongs. For example a socket config doesn't know if advanced network is enabled or it doesn't know if it's part of client or member network config. But I can move the method itself to `AltoConfig` class if you want. But the caller should be `TpcServerBootstrap` class.

Reply:
A good example of problematic design (lack of validation option on config) to spread to other parts of the system.